monkey/common/configuration/agent_sub_configuration_schemas.py

```diff
@@ -12,12 +14,48 @@
     TCPScanConfiguration,
 )
 
+valid_windows_custom_pba_filename_regex = re.compile(r"^[^<>:\"\\\/|?*]*[^<>:\"\\\/|?* \.]+$|^$")
+valid_linux_custom_pba_filename_regex = re.compile(r"^[^\\]*$")
```

Comment:
Since you're using a raw string (`r""`), I believe it's unnecessary to escape the backslash. You can test and verify.

```suggestion
valid_linux_custom_pba_filename_regex = re.compile(r"^[^\]*$")
```
That being said, we need to also prevent null bytes.
```suggestion
valid_linux_custom_pba_filename_regex = re.compile("^[^\0\\]*$")
```

Reply:
Oh, also, '\' is allowed in Linux file names. '/' is not.

Reply:
Still need to escape it

```
>>> re.compile(r"^[^\]*$")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\smalviya\AppData\Local\Programs\Python\Python37\lib\re.py", line 236, in compile
    return _compile(pattern, flags)
  File "C:\Users\smalviya\AppData\Local\Programs\Python\Python37\lib\re.py", line 288, in _compile
    p = sre_compile.compile(pattern, flags)
  File "C:\Users\smalviya\AppData\Local\Programs\Python\Python37\lib\sre_compile.py", line 764, in compile
    p = sre_parse.parse(p, flags)
  File "C:\Users\smalviya\AppData\Local\Programs\Python\Python37\lib\sre_parse.py", line 924, in parse
    p = _parse_sub(source, pattern, flags & SRE_FLAG_VERBOSE, 0)
  File "C:\Users\smalviya\AppData\Local\Programs\Python\Python37\lib\sre_parse.py", line 420, in _parse_sub
    not nested and not items))
  File "C:\Users\smalviya\AppData\Local\Programs\Python\Python37\lib\sre_parse.py", line 526, in _parse
    source.tell() - here)
re.error: unterminated character set at position 1
```

Updated to prevent null bytes.